File: tut04/client1.py
import socket

SERVER_HOST = '127.0.0.1'  # Replace with the server's IP address
SERVER_PORT = 12345  # Replace with the server's port

def main():
    try:
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client_socket:
            client_socket.connect((SERVER_HOST, SERVER_PORT))
            message = "Hello, IITP"
            client_socket.send(message.encode('utf-8'))

            response = client_socket.recv(1024).decode('utf-8')
            print("Received response from server:", response)
    except ConnectionRefusedError:
        print("Connection to the server refused.")
    except Exception as e:
        print("An error occurred:", str(e))

if __name__ == "__main__":
    main()

File: tut05/client1.cpp
#include <iostream>
#include <cstring>
#include <cstdlib>
#include <unistd.h>
#include <arpa/inet.h>
#include <sys/socket.h>

int main() {
    // Create a socket
    int clientSocket = socket(AF_INET, SOCK_STREAM, 0);
    if (clientSocket == -1) {
        perror("Error creating socket");
        exit(EXIT_FAILURE);
    }

    // Specify the server address and port
    struct sockaddr_in serverAddr;
    serverAddr.sin_family = AF_INET;
    serverAddr.sin_port = htons(8080);                   // Use the same port as the server
    serverAddr.sin_addr.s_addr = inet_addr("127.0.0.1"); // Replace with the server's IP address

    // Connect to the server
    if (connect(clientSocket, (struct sockaddr *)&serverAddr, sizeof(serverAddr)) == -1) {
        perror("Error connecting to server");
        exit(EXIT_FAILURE);
    }

    // Send a message to the server
    const char *message = "Hello IITP";
    int messageLength = strlen(message);
    int bytesSent = send(clientSocket, message, messageLength, 0);
    if (bytesSent == -1) {
        perror("Error sending data");
        exit(EXIT_FAILURE);
    }

    // Receive a response from the server
    char buffer[1024];
    int bytesRead = recv(clientSocket, buffer, sizeof(buffer) - 1, 0);
    if (bytesRead == -1) {
        perror("Error receiving data");
        exit(EXIT_FAILURE);
    }

    buffer[bytesRead] = '\0';
    printf("Received response from server: %s\n", buffer);

    close(clientSocket);

    return 0;
}
